src/lib/translators/mongo_dates.ts

```diff
@@ -179,3 +179,34 @@ export const ADVANCED_DATE_EXTRACT_MAP: Record<
     $isoWeek: { $subtract: [$$(step.column), 7 * 24 * 60 * 60 * 1000] },
   }),
 };
+
+type MongoStep = Record<string, any>;
+
+export const transformDateExtractFactory = (advancedDateExtractMap = ADVANCED_DATE_EXTRACT_MAP) => {
```

Comment:
Nice idea to have a factory. I personally found it easier to read to have protected methods or attributes that would get overridden, but it's just a matter of style.